Objects/dictobject.c

```diff
@@ -276,25 +276,35 @@ get_dict_freelist(void)
     struct _Py_object_freelists *freelists = _Py_object_freelists_GET();
     return &freelists->dicts;
 }
+
+static struct _Py_dictkeys_freelist *
+get_dictkeys_freelist(void)
+{
+    struct _Py_object_freelists *freelists = _Py_object_freelists_GET();
+    return &freelists->dictkeys;
+}
 #endif
 
 
 void
 _PyDict_ClearFreeList(struct _Py_object_freelists *freelists, int is_finalization)
 {
 #ifdef WITH_FREELISTS
-    struct _Py_dict_freelist *state = &freelists->dicts;
-    while (state->numfree > 0) {
-        PyDictObject *op = state->free_list[--state->numfree];
+    struct _Py_dict_freelist *dict_freelist = &freelists->dicts;
+    while (dict_freelist->numfree > 0) {
+        PyDictObject *op = dict_freelist->free_list[--dict_freelist->numfree];
         assert(PyDict_CheckExact(op));
         PyObject_GC_Del(op);
     }
-    while (state->keys_numfree > 0) {
-        PyMem_Free(state->keys_free_list[--state->keys_numfree]);
+    struct _Py_dictkeys_freelist *dictkeys_freelist = &freelists->dictkeys;
+    while (dictkeys_freelist->keys_numfree > 0) {
```

Comment:
As noted above.  I'd keep only the "numfree" field on the `_Py_dictkeys_freelist` struct.  The "keys_" prefix in "keys_numfree" is redundant.